maas/client/viscera/boot_resources.py

```diff
@@ -0,0 +1,285 @@
+
+
+class BootResourcesType(ObjectType):
+    """Metaclass for `BootResources`."""
+
+    def __iter__(cls):
+        return map(cls._object, cls._handler.read())
+
+
+class BootResources(ObjectSet, metaclass=BootResourcesType):
+    """The set of boot resources."""
+
+    @classmethod
+    def read(cls):
+        """Get list of `BootResource`'s.
+
+        Each `BootResource` from the API is limited in that it does not
+        contain the `sets` in the output. To get full data for each object
+        a read needs to be performed on each object::
+
+            resources = BootResources.read()
+            resources = [
+                BootResource.read(resource.id)
+                for resource in resources
+            ]
+
+        """
+        return cls(cls)
+
+    @classmethod
+    def start_import(cls):
+        """Start the import of `BootResource`'s."""
+        # Cannot use cls._handler.import() as import is a reserved statement.
+        return getattr(cls._handler, "import")()
+
+    @classmethod
+    def stop_import(cls):
+        """Stop the import of `BootResource`'s."""
+        return cls._handler.stop_import()
+
+    @classmethod
+    def create(
+            cls, name: str, architecture: str, content: io.IOBase, *,
+            title: str=None,
+            filetype: BootResourceFiletype=BootResourceFiletype.TGZ,
+            chunk_size=(1 << 22), progress_callback: Callable=None):
+        """Create a `BootResource`.
+
+        Creates an uploaded boot resource with `content`. The `content` is
+        uploaded in chunks of `chunk_size`. `content` must be seekable as the
+        first pass through the `content` will calculate the size and sha256
+        value then the second pass will perform the actual upload.
+
+        :param name: Name of the boot resource. Must be in format 'os/release'.
+        :type name: `str`
+        :param architecture: Architecture of the boot resource. Must be in
+            format 'arch/subarch'.
+        :type architecture: `str`
+        :param content: Content of the boot resource.
+        :type content: `io.IOBase`
+        :param title: Title of the boot resource.
+        :type title: `str`
+        :param filetype: Type of file in content.
+        :type filetype: `str`
+        :param chunk_size: Size in bytes to upload to MAAS in chunks.
+            (Default is 4 MiB).
+        :type chunk_size: `int`
+        :param progress_callback: Called to inform the current progress of the
+            upload. One argument is passed with the progress as a precentage.
+            If the resource was already complete and no content
+            needed to be uploaded then this callback will never be called.
+        :type progress_callback: Callable
+        :returns: Create boot resource.
+        :rtype: `BootResource`.
+        """
+        if not isinstance(name, str):
+            raise TypeError("name must be a str, not %s" % type(name).__name__)
```

Comment:
Ah I thought it was a testing only thing. I would prefer to not make it only in testing. I prefer it as always on.